utils/misc.ml

```diff
@@ -272,6 +272,32 @@ let repeated_label l =
 
 (** {1 Minimal support for Unicode characters in identifiers} *)
 
+
+module Uchar_more = struct
+
+  let string0 u =
+    let b = Buffer.create 10 in
+    Buffer.add_utf_8_uchar b u;
+    Buffer.contents b
+
+  let string u =
+    if Uchar.to_int u < 128  then begin
+      match Uchar.to_char u with
+      | '\n' -> {|\n|}
+      | '\'' -> {|\'|}
+      | '\\' -> {|\\|}
+      | '\r' -> {|\r|}
+      | '\b' -> {|\b|}
+      | '\t' -> {|\t|}
+      | c -> String.make 1 c
+    end
+    else string0 u
```

Comment:
If this is the uchar equivalent of `Char.escaped` (the function underlying the %C format), which I'm guessing it is based on usage, then I'd suggest renaming the function and implementation to match:

```suggestion
  let escaped u =
    if Uchar.to_int u < 128
    then Char.escaped (Uchar.to_char u)
    else string0 u
```

In the string0 branch, the code point will be displayed as is, without any escaping, which is probably not ideal. All kinds of whitespaces and separators chars will be displayed as `' '`. Do we have any unicode data that would allow us to distinguish characters that makes sense to display as in vs escaped?

Reply:
This would be possible but this requires to store a part of the unicode database.
I think it would be a good idea at a later point, I don't think this is required from the start.